src/app/component/component.component.ts

```diff
@@ -34,4 +36,16 @@ export class ComponentComponent implements OnInit {
       name.setValue(value, { emitEvent: false })
     });    
   }
+
+  public isObjectSchema() : boolean { 
```

Comment:
This seems like a smell to me although not too important right now. There are other schema types that will need an editor to open and I don't think having N `isX()` methods would be a good pattern. This can be refactored once we start adding more schema editor dialogs.

FYI, You also have this on `ObjectSchemaComponent`

Reply:
I'll move it inside the ObjectSchemaEditorService so it can be used across components. Can then just refactor from one place when we have a better pattern. I'd be curious how else you would do this?

Reply:
Well, you can you bind an expression on the view itself directly based on the value, or you could just have a single generic method wrapper that checks the type onChange(). This could be encapsulated into a drop down + schema editor component that itself could just be reused across other components.